tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -166,6 +166,61 @@ bool IsUnstridedSlice(const HloInstruction* hlo) {
                         [](int64 stride) { return stride == 1; });
 }
 
+// Returns bool to determine whether a pair of converts can be eliminated
+bool IsConvertPairNoOp(const HloInstruction* convert) {
+  //                       [operand_convert]          [convert]
+  // (operand_convert_input)->convert-(convert_input)->convert-(dest)
+
+  const HloInstruction* operand_convert = convert->operand(0);
+  Shape operand_convert_input_shape = operand_convert->operand(0)->shape();
+  Shape operand_convert_shape = operand_convert->shape();
```

Comment:
It might be slightly clearer to define & use a more obvious terminology here.  How about: `// src->intermediate->dst` and then `PrimitiveType src_type = ...; PrimitiveType intermediate_type = ...`,  etc?